### PR TITLE
Update character-set-and-collation.md

### PR DESCRIPTION
**Pull request by ireneontheway**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3402
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by nullnotnil** [comment]
Sorry, I can't review cn documents :-) @bb7133 PTAL

File: character-set-and-collation.md [hunk 1]
```diff
@@ -9,9 +9,41 @@ aliases: ['/docs-cn/dev/character-set-and-collation/','/docs-cn/dev/reference/sq
 
 ## 字符集和排序规则的概念
 
-字符集 (character set) 是符号与编码的集合。
+字符集 (character set) 是符号与编码的集合。TiDB 中的默认字符集是 utf8mb4，与 MySQL 8.0 及更高版本中的默认字符匹配。
```
Comment (yikeke) [inline character-set-and-collation.md:12]:
```suggestion
字符集 (character set) 是符号与编码的集合。TiDB 中的默认字符集是 utf8mb4，与 MySQL 8.0 及更高版本中的默认字符集匹配。
```

File: character-set-and-collation.md [hunk 2]
```diff
@@ -9,9 +9,41 @@ aliases: ['/docs-cn/dev/character-set-and-collation/','/docs-cn/dev/reference/sq
...
+mysql> SELECT 'A' = 'a';
++-----------+
+| 'A' = 'a' |
++-----------+
+|         1 |
++-----------+
+1 row in set (0.00 sec)
+```
+
+TiDB 默认使用二进制排序规则。与 MySQL 不同，MySQL 默认使用不区分大小写的排序规则。
```
Comment (yikeke) [inline character-set-and-collation.md:46]:
```suggestion
TiDB 默认使用二进制排序规则。这一点与 MySQL 不同，MySQL 默认使用不区分大小写的排序规则。
```

File: character-set-and-collation.md [hunk 3]
```diff
@@ -52,6 +84,22 @@ mysql> show collation;
 5 rows in set (0.01 sec)
 ```
 
+TiDB支持以下排序规则：
```
Comment (yikeke) [inline character-set-and-collation.md:87]:
```suggestion
TiDB 支持以下排序规则：
```

File: character-set-and-collation.md [hunk 4]
```diff
@@ -74,6 +122,42 @@ SHOW COLLATION WHERE Charset = 'utf8mb4';
 2 rows in set (0.00 sec)
 ```
 
+## TiDB中的`utf8`和`ut8mb4`
```
Comment (yikeke) [inline character-set-and-collation.md:125]:
```suggestion
## TiDB 中的 `utf8` 和 `ut8mb4`
```
基本的风格：记得中英文间空格

File: character-set-and-collation.md [hunk 5]
```diff
@@ -74,6 +122,42 @@ SHOW COLLATION WHERE Charset = 'utf8mb4';
...
+
+MySQL 限制字符集 `utf8` 为最多 3 个字节。这足以在基本多语言平面（BMP）中存储字符，但不足以存储表情符号等字符。因此，建议改用字符集`utf8mb4`。
```
Comment (yikeke) [inline character-set-and-collation.md:127]:
```suggestion
MySQL 限制字符集 `utf8` 为最多 3 个字节。这足以在基本多语言平面 (BMP) 中存储字符，但不足以存储表情符号等字符。因此，建议改用字符集`utf8mb4`。
```
见风格指南「括号」一节

File: character-set-and-collation.md [hunk 6]
```diff
@@ -74,6 +122,42 @@ SHOW COLLATION WHERE Charset = 'utf8mb4';
...
+
+默认情况下，TiDB同样限制字符集 `utf8` 为最多 3 个字节，以确保TiDB中创建的数据可以在 MySQL 中顺利恢复。可以在 TiDB 配置文件中将 `check-mb4-value-in-utf8` 的值更改为 `FALSE` 来禁用此功能。
```
Comment (yikeke) [inline character-set-and-collation.md:129]:
```suggestion
默认情况下，TiDB 同样限制字符集 `utf8` 为最多 3 个字节，以确保 TiDB 中创建的数据可以在 MySQL 中顺利恢复。你可以禁用此功能，方法是在 TiDB 配置文件中将 `check-mb4-value-in-utf8` 的值更改为 `FALSE`。
```

File: character-set-and-collation.md [hunk 7]
```diff
@@ -74,6 +122,42 @@ SHOW COLLATION WHERE Charset = 'utf8mb4';
...
+
+以下示例演示了在表中插入 4 字节的表情符号字符时的默认行为。`utf8` 字符集下`INSERT` 语句不能执行，`ut8mb4` 字符集下可以执行 `INSERT` 语句：
```
Comment (yikeke) [inline character-set-and-collation.md:131]:
```suggestion
以下示例演示了在表中插入 4 字节的表情符号字符时的默认行为。`utf8` 字符集下 `INSERT` 语句不能执行，`ut8mb4` 字符集下可以执行 `INSERT` 语句：
```

File: character-set-and-collation.md [hunk 8]
```diff
@@ -74,6 +122,42 @@ SHOW COLLATION WHERE Charset = 'utf8mb4';
 2 rows in set (0.00 sec)
 ```
 
+## TiDB 中的 `utf8` 和 `ut8mb4`
+
+MySQL 限制字符集 `utf8` 为最多 3 个字节。这足以在基本多语言平面 (BMP) 中存储字符，但不足以存储表情符号等字符。因此，建议改用字符集`utf8mb4`。
```
Comment (wjhuang2016) [inline character-set-and-collation.md:127]: 这足以存储在基本多语言平面 (BMP) 中的字符，但不足以存储表情符号等字符。

Comment (ireneontheway) [reply character-set-and-collation.md]: Done

**Pull request merged**